lib/pharos/phase_manager.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Pharos
+  class PhaseManager
+    include Pharos::Logging
+
+    # @param dirs [Array<String>]
+    def self.load_phases(*dirs)
+      dirs.each do |dir|
+        Dir.glob(File.join(dir, '*.rb')).each { |f| require(f) }
+      end
+    end
+
+    # @param dirs [Array<String>]
+    def initialize(ssh_manager:, **options)
+      @ssh_manager = ssh_manager
+      @options = options
+    end
+
+    # @param phases [Array<Pharos::Phases::Base>]
+    # @return [Array<...>]
+    def run_parallel(phases)
+      threads = phases.map { |phase|
+        Thread.new do
+          begin
+            yield phase
+          rescue StandardError => exc
+            puts " [#{phase}] #{exc.class}: #{exc.message}"
```

Comment:
If we get some better `Pharos::Logging`, then this should probably be something like `logger.error`... ATM it makes very little difference.

This is needed to report errors as soon as they happen, because it may take some time before the `thread.value` re-raises it.